### PR TITLE
Change return type for PytorchInferenceRunner

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
R: @yeandy @ryanthompson591

**Review by yeandy — APPROVED** [review]
LGTM

**Comment by AnandInguva** [comment]
CC: @tvalentyn @TheNeuralBit

File: sdks/python/apache_beam/ml/inference/pytorch.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(self, device: torch.device):
     self._device = device
 
   def run_inference(self, batch: List[torch.Tensor],
-                    model: torch.nn.Module) -> Iterable[torch.Tensor]:
+                    model: torch.nn.Module) -> Iterable[PredictionResult]:
     """
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/ml/inference/pytorch.py:41]: If the contract is that this should produce an iterable, I wonder if we should avoid materializing the list by returning a generator?

**Review by TheNeuralBit — APPROVED** [review]
Small, unrelated question. This LGTM

File: sdks/python/apache_beam/ml/inference/pytorch.py (hunk 1 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/pytorch.py:41]: I think its done in the later stage in a DoFn 

https://github.com/apache/beam/blob/68b9100db4b82196eba025bd417c46d03656f7c0/sdks/python/apache_beam/ml/inference/base.py#L210

Comment (TheNeuralBit) [reply sdks/python/apache_beam/ml/inference/pytorch.py:41]: What I'm suggesting is that we could avoid materializing a list between here and there by returning a generator (`(PredictionResult(x, y) for x, y in zip(batch, predictions))`) here. It looks like that would be undermined by the base DoFn materializing a list again here anyway: https://github.com/apache/beam/blob/68b9100db4b82196eba025bd417c46d03656f7c0/sdks/python/apache_beam/ml/inference/base.py#L198

Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/pytorch.py:41]: Ah got it.

**Pull request merged**